src/alcotest/alcotest_stubs.c

```diff
@@ -56,17 +52,24 @@ CAMLprim value ocaml_alcotest_get_terminal_dimensions(value unit)
 	CAMLreturn(result);
 }
 
-// POSIX support
-#elif defined(OCAML_ALCOTEST_POSIX)
+#elif defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
+#include <unistd.h>
 #include <sys/ioctl.h>
+#include <fcntl.h>
 
 CAMLprim value ocaml_alcotest_get_terminal_dimensions(value unit)
 {
 	CAMLparam1(unit);
 	CAMLlocal2(result, pair);
 	struct winsize ws;
-	int z = ioctl(STDOUT_FILENO, TIOCGWINSZ, &ws);
-	if (z == 0)
+	int fd = open("/dev/tty", O_RDONLY | O_NONBLOCK | O_CLOEXEC);
```

Comment:
In case someone else is looking at this and doesn't know why `STDOUT_FILENO` was replaced by `"/dev/tty"` (like I was), the difference is when the output of alcotest is piped (say through a grep): stdout wouldn't have a size so the formatting would be different (and probably be confusing to users?), while the tty always return the same size (... I don't have strong opinion regarding this change, in general it's weird for piped programs to respect terminal size but it seems fine for alcotest)